src/libraries/System.Runtime.InteropServices/tests/System.Runtime.InteropServices.UnitTests/System/Runtime/InteropServices/ComVariantTests.cs

```diff
@@ -0,0 +1,493 @@
+            Assert.Throws<ArgumentException>("T", () => variant.As<Blob>());
+            Assert.Equal(blob, variant.GetRawDataRef<Blob>());
+        }
+
+        [Fact]
+        public void Stream_Raw()
+        {
+            IntPtr nativeStream = 42;
+            // Using a fake value so we aren't disposing the ComVariant instance.
+            ComVariant variant = ComVariant.CreateRaw(VarEnum.VT_STREAM, nativeStream);
+            Assert.Equal(VarEnum.VT_STREAM, variant.VarType);
+            Assert.Equal(nativeStream, variant.GetRawDataRef<IntPtr>());
+        }
+
+        [Fact]
+        public void Storage_Raw()
+        {
+            IntPtr nativeStorage = 42;
+            // Using a fake value so we aren't disposing the ComVariant instance.
+            ComVariant variant = ComVariant.CreateRaw(VarEnum.VT_STORAGE, nativeStorage);
+            Assert.Equal(VarEnum.VT_STORAGE, variant.VarType);
+            Assert.Equal(nativeStorage, variant.GetRawDataRef<IntPtr>());
+        }
+
+        [Fact]
+        public void StreamedObject_Raw()
+        {
+            IntPtr nativeStream = 42;
+            // Using a fake value so we aren't disposing the ComVariant instance.
+            ComVariant variant = ComVariant.CreateRaw(VarEnum.VT_STREAMED_OBJECT, nativeStream);
+            Assert.Equal(VarEnum.VT_STREAMED_OBJECT, variant.VarType);
+            Assert.Equal(nativeStream, variant.GetRawDataRef<IntPtr>());
+        }
+
+        [Fact]
+        public void StoredObject_Raw()
+        {
+            IntPtr nativeStorage = 42;
+            // Using a fake value so we aren't disposing the ComVariant instance.
+            ComVariant variant = ComVariant.CreateRaw(VarEnum.VT_STORED_OBJECT, nativeStorage);
+            Assert.Equal(VarEnum.VT_STORED_OBJECT, variant.VarType);
+            Assert.Equal(nativeStorage, variant.GetRawDataRef<IntPtr>());
+        }
+
+        [Fact]
+        public void VersionedStream_Raw()
+        {
+            IntPtr nativeStream = 42;
+            // Using a fake value so we aren't disposing the ComVariant instance.
+            ComVariant variant = ComVariant.CreateRaw((VarEnum)73, nativeStream);
+            Assert.Equal((VarEnum)73, variant.VarType);
+            Assert.Equal(nativeStream, variant.GetRawDataRef<IntPtr>());
+        }
+
+        [Fact]
+        public void BlobObject_Raw()
+        {
+            Blob blob = new Blob { Length = 3, Data = Marshal.AllocCoTaskMem(10) };
+            using ComVariant variant = ComVariant.CreateRaw(VarEnum.VT_BLOB_OBJECT, blob);
+            Assert.Equal(VarEnum.VT_BLOB_OBJECT, variant.VarType);
+            Assert.Throws<ArgumentException>("T", () => variant.As<Blob>());
+            Assert.Equal(blob, variant.GetRawDataRef<Blob>());
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        private unsafe struct ClipboardData
+        {
+            public uint _size;
+            public int _format;
+            public IntPtr _data;
```

Comment:
I believe this is supposed to be an offset into the `ClipboardData` instance. Similar to the C style:

```c
struct ClipboardData
{
    uint32_t size;
    int32_t format;
    uint8_t data[];
};
```

Reply:
The definition in the Windows headers is the following:

```c
typedef struct tagCLIPDATA
    {
    ULONG cbSize;
    LONG ulClipFmt;
    /* [size_is] */ BYTE *pClipData;
    } 	CLIPDATA;
```

Reply:
Okay, looks good then.